### PR TITLE
Update desktop-test-app and template to use iui-layouts

### PR DESCRIPTION
**Pull request by AnDuong249**
- Update the desktop-test-app and the cra-template-desktop-viewer to use itwinui-layouts-react
- Fix css for SelectItwin component
- Replaced LabeledInput with SearchBox in SelectItwin.tsx
- Clean up unnecessary css

Before:image
After: image

Before:image
After: image

Before:
image
After: image

### CODE REVIEW

File: packages/apps/desktop-viewer-test/src/frontend/components/routes/ViewerRoute.tsx [hunk 1]
```diff
@@ -3,12 +3,15 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 
+import "@itwin/itwinui-layouts-css/styles.css";
```
Comment (aruniverse) [inline packages/apps/desktop-viewer-test/src/frontend/components/routes/ViewerRoute.tsx:6]: this file should only need to be imported once, at the root of the project

**Comment by aruniverse** [comment]
this pr, can still use some work; take a look at this layout: https://itwin.github.io/iTwinUI-layouts/#/page-layout-2

also, fix the ci failure coming from i imagine is the audit failure

**Comment by AnDuong249** [comment]
> this pr, can still use some work; take a look at this layout: https://itwin.github.io/iTwinUI-layouts/#/page-layout-2
> 
> also, fix the ci failure coming from i imagine is the audit failure

Should I make a separate PR to bump it like Ben did in core yesterday?

File: packages/apps/desktop-viewer-test/src/frontend/components/App.tsx [hunk 2]
```diff
@@ -3,9 +3,12 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 
+import "@itwin/itwinui-layouts-css/styles.css";
```
Comment (aruniverse) [inline packages/apps/desktop-viewer-test/src/frontend/components/App.tsx:6]: id recommend adding this at the desktop frontend root, which happens to be `frontend/index.tsx`

File: packages/apps/desktop-viewer-test/src/frontend/components/modelSelector/SelectIModel.tsx [hunk 3]
```diff
@@ -199,9 +200,10 @@ export const SelectIModel = ({
 
   return (
     <div className="itv-scrolling-container select-imodel">
-      <div className={"itv-content-margins"}>
+      <PageLayout.TitleArea className={"itv-content-margins"}>
```
Comment (aruniverse) [inline packages/apps/desktop-viewer-test/src/frontend/components/modelSelector/SelectIModel.tsx:203]: so is this PageLayout.TitleArea inside a PageLayout.Content ?
does that look right? should it be a parent/child or siblings? 

maybe it looks fine. i havent pulled, but just wondering based off your code

also, i think might be nice to add a title area for the select project pg to say something about what itwin/projects imodels are being shown

File: packages/apps/desktop-viewer-test/src/frontend/components/modelSelector/SelectIModel.tsx [hunk 4]
```diff
@@ -199,9 +200,10 @@ export const SelectIModel = ({
 
   return (
     <div className="itv-scrolling-container select-imodel">
```
Comment (aruniverse) [inline packages/apps/desktop-viewer-test/src/frontend/components/modelSelector/SelectIModel.tsx:202]: are any of these custom classes we created being used anymore / still relevant with the introduction of the layout pkg?

File: packages/modules/desktop-viewer-react/package.json [hunk 5]
```diff
@@ -78,7 +78,7 @@
     "@itwin/core-frontend": "^4.0.0",
     "@itwin/ecschema-metadata": "^4.0.0",
     "@itwin/electron-authorization": "^0.13.0",
-    "electron": "^24.0.0",
+    "electron": "^24.8.3",
```
Comment (aruniverse) [inline packages/modules/desktop-viewer-react/package.json:81]:
```suggestion
    "electron": "^24.0.0",
```

updating peers is breaking. we just need to update the dev dep in this pkg

File: common/changes/@itwin/desktop-viewer-react/AddItwinUILayouts_2023-09-19-23-21.json [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{
+  "changes": [
+    {
+      "packageName": "@itwin/desktop-viewer-react",
+      "comment": "CVE-2023-4863 - bump electron to 24.8.3",
```
Comment (aruniverse) [inline common/changes/@itwin/desktop-viewer-react/AddItwinUILayouts_2023-09-19-23-21.json:5]:
```suggestion
      "comment": "",
```

File: packages/apps/desktop-viewer-test/src/frontend/components/App.tsx [hunk 7]
```diff
@@ -85,12 +86,30 @@ const App = () => {
...
+                }
+              >
+                <Route path="/" element={<HomeRoute />} />
+                <Route path="/itwins/:iTwinId" element={<IModelsRoute />} />
+                <Route path="/itwins" element={<ITwinsRoute />} />
+              </Route>
+              <Route
+                element={
+                  <PageLayout.Content>
+                    <Outlet />
```
Comment (AnDuong249) [inline packages/apps/desktop-viewer-test/src/frontend/components/App.tsx:105]: Unfortunately react-router doesn't allow you to wrap individual Routes with something so this is the only way I can think of.
https://reactrouter.com/en/main/start/concepts#layout-routes

**Review by aruniverse — APPROVED** [review]
great work!

**Pull request merged**